SeaPkg/Core/Init/StmInit.c

```diff
@@ -659,6 +665,27 @@ BspInit (
   IdtGate                       = (IA32_IDT_GATE_DESCRIPTOR *)mHostContextCommon.Idtr.Base;
   InitializeExternalVectorTablePtr (IdtGate);
 
+  //
+  // Allocate XState buffer
+  //
+  XStateSize                                 = CalculateXStateSize ();
+  mGuestContextCommonNormal.ZeroXStateBuffer = (UINTN)AllocatePages (STM_SIZE_TO_PAGES (XStateSize));
+  for (SubIndex = 0; SubIndex < mHostContextCommon.CpuNum; SubIndex++) {
+    mGuestContextCommonNormal.GuestContextPerCpu[SubIndex].XStateBuffer = (UINTN)AllocatePages (STM_SIZE_TO_PAGES (XStateSize));
+  }
+
+  for (SubIndex = 0; SubIndex < mHostContextCommon.CpuNum; SubIndex++) {
+    mHostContextCommon.HostContextPerCpu[SubIndex].HostMsrEntryCount          = 1;
+    mGuestContextCommonNormal.GuestContextPerCpu[SubIndex].GuestMsrEntryCount = 1;
+  }
+
+  mHostContextCommon.HostContextPerCpu[0].HostMsrEntryAddress          = (UINT64)(UINTN)AllocatePages (STM_SIZE_TO_PAGES (sizeof (VM_EXIT_MSR_ENTRY) * mHostContextCommon.HostContextPerCpu[0].HostMsrEntryCount * mHostContextCommon.CpuNum));
+  mGuestContextCommonNormal.GuestContextPerCpu[0].GuestMsrEntryAddress = (UINT64)(UINTN)AllocatePages (STM_SIZE_TO_PAGES (sizeof (VM_EXIT_MSR_ENTRY) * mGuestContextCommonNormal.GuestContextPerCpu[0].GuestMsrEntryCount * mHostContextCommon.CpuNum));
+  for (SubIndex = 0; SubIndex < mHostContextCommon.CpuNum; SubIndex++) {
+    mHostContextCommon.HostContextPerCpu[SubIndex].HostMsrEntryAddress          = mHostContextCommon.HostContextPerCpu[0].HostMsrEntryAddress + sizeof (VM_EXIT_MSR_ENTRY) * mGuestContextCommonNormal.GuestContextPerCpu[0].GuestMsrEntryCount * SubIndex;

```

Comment:
Shouldn't this be all in the Host Context?  Using the Guest context seems incorrect.

Reply:
I see that the highlighted part didn't come through when I left the comment.  Line 685 the guest and host context are both being used.

Reply:
I agree that the context used should be consistent.

Reply:
Updated